edgeless_telemetry/src/elasticsearch_resources.rs

```diff
@@ -0,0 +1,71 @@
+use serde::{Deserialize, Serialize};
+use serde_json::{json, Value};
+// use sysinfo::System::SystemExt;
+use chrono::{DateTime, Utc};
+use std::sync::{Arc, Mutex};
+use std::thread;
+use std::time::Duration;
+use sysinfo::*;
+
+use crate::elasticsearch_api;
+
+#[derive(Debug, Serialize, Deserialize)]
+struct SystemResources {
+    cpu_percent: f32,
+    memory_percent: f32,
+    timestamp: DateTime<Utc>,
+}
+
+fn convert_to_value(data: &SystemResources) -> Value {
+    json!({
+        "cpu_percent": data.cpu_percent,
+        "memory_percent": data.memory_percent,
+        "timestamp": data.timestamp,
+    })
+}
+
+#[tokio::main]
+async fn elasticsearch_resources() {
+    let system = Arc::new(Mutex::new(System::new_all()));
+
+    //periodically update the system information
+    let system_clone = Arc::clone(&system);
+    thread::spawn(move || loop {
```

Comment:
we don't use threads but tokio tasks.

Reply:
if you really would block the async coroutine you might need a thread, but that is also handled differently.

Reply:
Also, why is this done separately from using the values?